app.rb

```diff
@@ -0,0 +1,112 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'books'
+require_relative 'rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'List of all books:'
+    @books.each { |book| puts "#{book.title} written by: #{book.author}" }
+  end
+
+  def list_all_people
+    puts 'List of people:'
+    @people.each do |person|
+      if person.is_a?(Student)
+        puts "[Student] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}, Classroom: #{person.classroom}"
+      elsif person.is_a?(Teacher)
+        puts "[Teacher] ID: #{person.id}, Name: #{person.name}, Age: #{person.age},
+        Specialization: #{person.specialization}"
+      else
+        puts 'Invalid either student or teacher please'
+      end
+    end
+  end
+
+  def create_a_person # rubocop:todo Metrics/MethodLength
+    print 'Enter your name: '
+    name = gets.chomp
+    print 'Enter your Age: '
+    age = gets.chomp.to_i
+
+    print "Are you a student or a teacher? (Enter 'student' or 'teacher'): "
+    role = gets.chomp.downcase
+    puts "You are a : #{role}"
+
+    if role == 'student'
+      print 'Classroom: '
+      classroom = gets.chomp
+      person = Student.new(classroom: classroom, name: name, age: age)
+    elsif role == 'teacher'
+      print 'Specialization: '
+      specialization = gets.chomp
+      person = Teacher.new(name: name, age: age, specialization: specialization)
+    else
+      puts 'Invalid role'
+      return
+    end
+    @people << person
+    puts "#{role.capitalize} #{name} created!"
+  end
+
+  def create_book
+    puts 'Enter attributes for your book:'
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts "Book '#{title}' by '#{author}' created."
+  end
+
+  def create_rental # rubocop:todo Metrics/MethodLength
+    puts 'Rental details:'
+    list_all_people
+    print 'Enter a person ID: '
+    person_id = gets.chomp.to_i
+    person = @people.find { |p| p.id == person_id }
+    if person.nil?
+      puts 'Person is not found.'
+      return
+    end
+    list_all_books
+    print 'Enter the book Title: '
+    book_title = gets.chomp
+    book = @books.find { |b| b.title == book_title }
+    if book.nil?
+      puts 'Book not found.'
+      return
+    end
+    print 'Enter rental Date (YYYY-MM-DD): '
+    date = gets.chomp
+    rental = Rental.new(date, book, person)
+    @rentals << rental
+    puts "Rental created for #{person.name} - #{book.title} on #{date}"
+  end
```

Comment:
- [ ] Hey! 🎬 Could you please take a peek at this [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) once more? 🧐 While I appreciate your fantastic work, the current user experience for creating rentals seems a bit challenging. 📚 Typing the title of a book might be cumbersome. 😓 Following the video's example, let's enhance the process by displaying lists of people and books with corresponding indices. 📋 This way, users can simply choose the index of the person and the index of the book they want when creating a rental, making the overall experience much smoother and user-friendly! 🚀

```suggestion
  def create_rental
    puts "Select a book from the following list by number"
    @books.each_with_index do |book, index|
      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
    end
    # *If needed, ask for parameters for the option.
    book_index = gets.chomp.to_i

    puts "\nSelect a person from the following list by number (not id)"
    @people.each_with_index do |person, index|
      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
    end
    # *If needed, ask for parameters for the option.
    person_index = gets.chomp.to_i

    print "\nDate: "
    # *If needed, ask for parameters for the option.
    date = gets.chomp

    @rentals << Rental.new(date, @books[book_index], @people[person_index])
    puts "Rental created successfully"
  end
```